pkg/schedule/schedulers/balance_range.go

```diff
@@ -70,13 +77,59 @@ type balanceRangeSchedulerConfig struct {
 type balanceRangeSchedulerJob struct {
 	JobID   uint64          `json:"job-id"`
 	Role    Role            `json:"role"`
-	Engine  string          `json:"engine"`
+	Engine  engine          `json:"engine"`
 	Timeout time.Duration   `json:"timeout"`
 	Ranges  []core.KeyRange `json:"ranges"`
 	Alias   string          `json:"alias"`
+	Start   *time.Time      `json:"start,omitempty"`
+	Finish  *time.Time      `json:"finish,omitempty"`
+	Create  time.Time       `json:"create"`
 	Status  JobStatus       `json:"status"`
 }
 
+func (conf *balanceRangeSchedulerConfig) begin(job *balanceRangeSchedulerJob) bool {
+	conf.Lock()
+	defer conf.Unlock()
+	if job.Status != pending {
+		return false
+	}
+	now := time.Now()
+	job.Start = &now
+	job.Status = running
+	if err := conf.save(); err != nil {
```

Comment:
If persist fails, do we still regard it as a running job?

Reply:
fixed, it needs to roll back.